lib/ASTGen/Sources/ASTGen/Exprs.swift

```diff
@@ -309,28 +309,47 @@ extension ASTGenVisitor {
       inLoc: signature.inLoc
     )
 
-    let body = self.withDeclContext(expr.asDeclContext) {
-      BridgedBraceStmt.createParsed(
-        self.ctx,
-        lBraceLoc: self.generateSourceLoc(node.leftBrace),
-        elements: self.generate(codeBlockItemList: node.statements),
-        rBraceLoc: self.generateSourceLoc(node.rightBrace)
-      )
-    }
-
     if signature.params == nil {
-      // TODO: Handle doller identifiers inside the closure.
       let loc = self.generateSourceLoc(node.leftBrace)
-      let params = BridgedParameterList.createParsed(
+      var params: [BridgedParamDecl] = []
+      if let anonymousParamMaxIndex = findMaxAnonymousClosureParamUsage(closureExpr: node) {
+        for idx in 0...anonymousParamMaxIndex {
+          let param = BridgedParamDecl.createParsed(
+            self.ctx,
+            declContext: expr.asDeclContext,
+            specifierLoc: nil,
+            argName: nil,
+            argNameLoc: nil,
+            paramName: ctx.getDollarIdentifier(idx),
+            paramNameLoc: loc,
+            type: nil,
+            defaultValue: nil
+          )
+          param.setSpecifier(.default)
+          param.setImplicit()
```

Comment:
IMO it would be nicer to expose a `createImplicit` method, but I don't feel too strongly about it